pkg/resources/task.go

```diff
@@ -348,24 +349,11 @@ func CreateTask(d *schema.ResourceData, meta interface{}) error {
 		precedingTasks := make([]sdk.SchemaObjectIdentifier, 0)
 		for _, dep := range after {
 			precedingTaskId := sdk.NewSchemaObjectIdentifier(databaseName, schemaName, dep)
-			rootTasks, err := sdk.GetRootTasks(client.Tasks, ctx, precedingTaskId)
+			resumeSuspended, err := client.Tasks.TemporarilySuspendRootTasks(ctx, precedingTaskId, taskId)
+			defer func() { returnedErr = errors.Join(returnedErr, resumeSuspended()) }()
 			if err != nil {
 				return err
 			}
-			for _, rootTask := range rootTasks {
-				// if a root task is started, then it needs to be suspended before the child tasks can be created
-				if rootTask.IsStarted() {
-					err := suspendTask(ctx, client, rootTask.ID())
-					if err != nil {
-						return err
-					}
-
-					// resume the task after modifications are complete as long as it is not a standalone task
```

Comment:
I don't get that part TBH. Why this is compared with the name of the task? (I ask it there because you have moved this logic to the SDK too).